tests/hipcc_check.cpp

```diff
@@ -10,8 +10,26 @@ int main() {
   constexpr int blocksize = 64;
   constexpr int size = gridsize * blocksize;
   int *d_buf;
-  hipMalloc(&d_buf, size * sizeof(int));
+  hipHostMalloc(&d_buf, size * sizeof(int));
   hipLaunchKernelGGL(squares, gridsize, blocksize, 0, 0, d_buf);
   hipDeviceSynchronize();
+
+  // Check results.
+  int mismatches_count = 0;
+  bool failed = false;
+  for (int i = 0; i < size; ++i) {
+    int square = i * i;
+    if (d_buf[i] != square) {
+      fprintf(stderr,
+              "Element at index %d expected value %d, actual value: %d\n", i,
```

Comment:
do we need to output the grid and blocksize? just maybe to better understand where within the calculation the error occurs

Reply:
With this simple program the results are _probably_ going to be all or nothing. Windows was getting all 0s since the kernel launch was silently failing. I figured we might as well check all the values just in case a value in the middle or at the edges is off.